lite2/client_test.go

```diff
@@ -904,3 +904,95 @@ func TestClientTrustedValidatorSet(t *testing.T) {
 	assert.NotNil(t, valSet)
 	assert.EqualValues(t, 2, height)
 }
+
+//
+// #################################  BENCHMARKING ######################################
+//
+
+// NOTE: block is produced every minute. Make sure the verification time provided in the function call is correct for
+// the size of the blockchain. The benchmarking may take some time hence it can be more useful to set the time or
+// the amount of iterations use the flag -benchtime t -> i.e. -benchtime 5m or -benchtime 100x
+// Remember that none of these benchmarks account for network latency
+var (
+	largeFullNode = mockp.New(GenMockNode(chainID, 1000, 100, 1, bTime))
+	result        *Client
+)
+
+// Creates a client using the largeFullNode
+func defaultClient() *Client {
+	genesisHeader, _ := largeFullNode.SignedHeader(1)
+	c, _ := NewClient(
+		chainID,
+		TrustOptions{
+			Period: 24 * time.Hour,
+			Height: 1,
+			Hash:   genesisHeader.Hash(),
+		},
+		largeFullNode,
+		[]provider.Provider{largeFullNode},
+		dbs.New(dbm.NewMemDB(), chainID),
+		Logger(log.TestingLogger()),
+	)
+	return c
+}
+
+func BenchmarkSequence(b *testing.B) {
+	c := defaultClient()
+	trustedHeader, _, err := c.fetchHeaderAndValsAtHeight(1)
+	require.NoError(b, err)
+	untrustedHeader, untrustedVals, err := c.fetchHeaderAndValsAtHeight(0)
+	require.NoError(b, err)
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		_ = c.sequence(trustedHeader, untrustedHeader, untrustedVals, bTime.Add(1000*time.Minute))
```

Comment:
can't we do this with public API?

Reply:
I guess I just wanted to test just the pure verification method not the entire thing - but I could change it so it uses `VerifyHeaderAtHeight()` instead